library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -793,6 +793,122 @@ void callSuperNotifyItemMoved(int fromPosition, int toPosition) {
         super.notifyItemMoved(fromPosition, toPosition);
     }
 
+    /**
+     * Helper method that calls {@link #notifyItemChanged} when section changed from
+     * LOADING/ FAILED/ EMPTY to LOADING/ FAILED/ EMPTY.
+     *
+     * @param tag unique identifier of the section
+     * @param previousState previous state of section
+     */
+    public void notifyNotLoadedStateChanged(String tag, Section.State previousState) {
+        Section section = getValidSectionOrThrowException(tag);
+
+        notifyNotLoadedStateChanged(section, previousState);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemChanged} when section changed from
+     * LOADING/ FAILED/ EMPTY to LOADING/ FAILED/ EMPTY.
+     *
+     * @param section a visible section of this adapter
+     * @param previousState previous state of section
+     */
+    public void notifyNotLoadedStateChanged(Section section, Section.State previousState) {
+        Section.State state = section.getState();
+
+        if (state == previousState) {
+            throw new IllegalStateException("No state changed");
+        }
+
+        if (previousState == Section.State.LOADED) {
+            throw new IllegalStateException("Use notifyStateChangedFromLoaded");
+        }
+
+        if (state == Section.State.LOADED) {
+            throw new IllegalStateException("Use notifyStateChangedToLoaded");
+        }
+
+        notifyItemChangedInSection(section, 0);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemChanged} and {@link #notifyItemInserted} when
+     * section changed from LOADING/ FAILED/ EMPTY to LOADED
+     *
+     * @param tag unique identifier of the section
+     * @param previousState previous state of section
+     */
+    public void notifyStateChangedToLoaded(String tag, Section.State previousState) {
+        Section section = getValidSectionOrThrowException(tag);
+
+        notifyStateChangedToLoaded(section, previousState);
+    }
+
+    /**
+     * Helper method that calls {@link #notifyItemChanged} and {@link #notifyItemInserted} when
+     * section changed from LOADING/ FAILED/ EMPTY to LOADED
+     *
+     * @param section a visible section of this adapter
+     * @param previousState previous state of section
+     */
+    public void notifyStateChangedToLoaded(Section section, Section.State previousState) {
+        Section.State state = section.getState();
+
+        if (state == previousState) {
+            throw new IllegalStateException("No state changed");
+        }
+
+        if (state != Section.State.LOADED) {
+            if (previousState == Section.State.LOADED) {
+                throw new IllegalStateException("Use notifyStateChangedFromLoaded");
+            } else {
+                throw new IllegalStateException("Use notifyNotLoadedStateChanged");
+            }
+        }
+
+        int contentItemsTotal = section.getContentItemsTotal();
+
+        notifyItemChangedInSection(section, 0);
```

Comment:
please invert the order of this line and the above - just a matter of code cohesion, no side effect